### ISSUE TITLE
DOIs on multiple lines are split up

### ISSUE THREAD

**Issue by michamos** [inspirehep/refextract#14]
We noticed with @kaplun that a DOI is sometimes broken up by TeX and appears on multiple lines. This is default behavior of the `url` LaTeX package (which is often used to typeset URLs and used internally by the `hyperref` package), contrarily to what @tsgit said earlier. In that case, only the part of the DOI on the first line is taken as a DOI, which is of course wrong.

The default behavior (in the `\UrlBreaks` and `\UrlBigBreaks` macros in [url.sty](https://www.ctan.org/tex-archive/macros/latex/contrib/url/url.sty)) is to allow line breaks after:
`. @ \ / ! _ | ; > ] ) , ? & ' + = # :` and also after `-` if the [hyphens] option is passed to the package.

**Comment by michamos** [comment]
Possible solution:

1. if the DOI ends in some of those characters (to be specified), have refextract spit out both the naive DOI and the same with the next word appended, recursively, as possible DOIs
2. call bibrank on all those DOIs that are substrings of each other, trying to match first the longest (to avoid the proceedings vs conference paper issue) and put the first match in the right MARC field
3. if there is no match, try to resolve the DOIs to find the right one

I don't know if this is compatible with the way refextract and bibrank talk to each other

**Comment by tsgit** [comment]
I stand corrected, tex does insert line breaks in dois when necessary in most circumstances

**Comment by kaplun** [comment]
@michamos, was the case we were looking today on arXiv? In that case we can look-up the sources and better understand the circumstances.

**Comment by michamos** [comment]
@kaplun yes, we were looking at an arXiv paper with this problem the other day. I don't really understand your remark though

**Comment by kaplun** [comment]
Nevermind: I misread @tsgit's "I stand corrected"... I.e. we all agree that LaTeX can indeed break on newline DOIs.
